roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,269 @@
+- Top level responsibilities of the scheduler
+- Interfaces
+- Considerations async backing (see ticket)
+
+# Interfaces for parathreads
+
+## User Interface
+
+The entry point for the user is twofold:
+
+1. Registering a PVF with some deposit
+2. Actually bidding for a slot.
+
+### Registering a PVF with some deposit
+
+This can very likely be equal or very similar to what we have for parachains.
+
+TODO: Figure out how this works for parachains or whether there even is a
+separate deposit for the PVF. I believe parachains just have locked DOTs for
+their lease period, which also covers PVF storage.
+
+### Bidding for a slot/Auctioning System
+
+#### Goals
+
+1. Price finding
+2. Be efficient ... we are talking about slots for single blocks, the overhead
+   of assignment should be minimal.
+3. Maximize utilization: Avoid empty slots, as long as there is demand from
+   parathread willing to pay the minimum price.
+4. DoS resistance: We should not be able to get flooded with dubious bids, that
```

Comment:
What is a dubious bid in this case? as long as I have registered some PVF, and I pay for it, what could be considered dubious?

Reply:
That would not be dubious. That's a very generic goal: It just means, well that we have to make sure someone gets charged and on the node side that we should be able to filter and disconnect peers sending garbage. (E.g. signed by an invalid account)